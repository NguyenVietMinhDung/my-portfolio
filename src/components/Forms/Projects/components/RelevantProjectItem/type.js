export type Props = {
  name: string,
  role: string,
  startDate: string,
  endDate: ?string,
  description: string,
};
