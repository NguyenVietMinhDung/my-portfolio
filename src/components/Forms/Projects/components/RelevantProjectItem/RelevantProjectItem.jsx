// @flow
import React, { memo } from 'react';
import CSSModules from 'react-css-modules';
import styles from './RelevantProjectItem.scss';
import type { Props } from './type';
import utils from '../../../../../utils';

const RelevantProjectItem = (props: Props) => {
  const {
    name, role, description, startDate, endDate,
  } = props;
  const { date: { getDuration } } = utils;
  const duration = getDuration(startDate, endDate);
  return (
    <li styleName="relevant-project-item">
      <div
        data-aos="zoom-in"
        data-aos-duration="2000"
        data-aos-offset="100"
        styleName="name"
      >
        {name}
      </div>
      <div
        data-aos="zoom-in"
        data-aos-duration="2000"
        data-aos-offset="100"
        styleName="summary"
      >
        <div styleName="description">{description}</div>
        <div styleName="role">{role}</div>
        <div styleName="duration">
          <i className="lni-calendar" />
          &nbsp;&nbsp;{duration}
        </div>
      </div>
    </li>
  );
};

export default memo<Props>(CSSModules(RelevantProjectItem, styles));
