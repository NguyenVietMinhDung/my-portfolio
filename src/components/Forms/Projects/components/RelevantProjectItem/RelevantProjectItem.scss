@import '../../../../../styles/abstracts/variables';
@import '../../../../../styles/abstracts/mixins';

.relevant-project-item {
	display: flex;

	margin-bottom: 50px;
	align-items: flex-start;
	flex-flow: row wrap;

	.name {
		position: relative;

		display: flex;

		width: 120px;
		height: 72px;
		align-items: center;
		flex-flow: row wrap;
		justify-content: center;

		@include font(inherit, inherit, $black-color, 14px, 500);

		&::after {
			position: absolute;
			top: 0;
			left: 0;

			width: 120px;
			height: 72px;

			content: '';

			background: url($mac-img) no-repeat center;
			background-size: cover;
		}
	}

	.summary {
		flex: 1;

		padding-left: 20px;

		.description {
			padding-bottom: 5px;
		}

		.role {
			@include font(inherit, inherit, $black-color, 14px, 500);
			padding-bottom: 10px;
		}

		.duration {
			@include font($default-font-family, inherit, inherit, 14px, 100);
		}
	}
}
