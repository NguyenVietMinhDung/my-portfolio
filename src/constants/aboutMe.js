export const TITLE = 'About Me';
export const GREETING = 'Hi There! I\'m Dung Nguyen';
export const SUMMARY = 'I am an ambitious Web\n'
  + 'Application Developer who\n'
  + 'always tries to improve himself\n'
  + 'everyday, a quick learner who\n'
  + 'can absorb new ideas and can\n'
  + 'communicate clearly and\n'
  + 'effectively. Looking for an\n'
  + 'ambitious company which will\n'
  + 'challenge my developer and\n'
  + 'allows me to continue\n'
  + 'to develop my knowledge and\n'
  + 'potential.';
export const basicInfo = [
  {
    iconClassName: 'lni-calendar',
    name: 'Date of birth',
    value: '16 January 1995',
  },
  {
    iconClassName: 'lni-envelope',
    name: 'Email',
    value: 'ngvietminhdung@gmail.com',
  },
  {
    iconClassName: 'lni-phone-handset',
    name: 'Phone',
    value: '(076) 401-2912',
  },
  {
    iconClassName: 'lni-map-marker',
    name: 'Location',
    value: 'Tan Binh District, HCM City',
  },
];
export const skills = [
  {
    itemId: 'skill-1',
    name: 'JavaScript, HTML, CSS',
    value: 80,
  },
  {
    itemId: 'skill-2',
    name: 'ReactJS',
    value: 60,
  },
  {
    itemId: 'skill-3',
    name: 'SCSS',
    value: 60,
  },
  {
    itemId: 'skill-5',
    name: 'Webpack',
    value: 50,
  },
  {
    itemId: 'skill-6',
    name: 'Node.js',
    value: 60,
  },
  {
    itemId: 'skill-7',
    name: 'Git',
    value: 60,
  },
];
