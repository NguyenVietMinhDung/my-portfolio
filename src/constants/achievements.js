export const TITLE = 'Achievements';
export const timeline = {
  iconClassName: 'lni-certificate',
  items: [
    {
      key: 1,
      organization: 'Amazon Web Services (AWS)',
      startDate: 'August 2020',
      endDate: 'August 2023',
      description: 'AWS Certified Cloud Practitioner',
    },
  ],
};
