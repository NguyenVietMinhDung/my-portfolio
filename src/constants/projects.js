export const TITLE = 'Relevant Projects';
export const items = [
  {
    isCurrentProject: false,
    name: 'Textme',
    role: 'Developer',
    startDate: 'September 2019',
    endDate: 'May 2020',
    description: 'A chat tool that helps customers can contact with bankers and versa vice',
  },
  {
    isCurrentProject: false,
    name: 'Huddle',
    role: 'Developer',
    startDate: 'January 2019',
    endDate: 'June 2019',
    description: 'A ticket-selling website for US high school events',
  },
  {
    isCurrentProject: false,
    name: 'ERTPMTool',
    role: 'Developer',
    startDate: 'July 2018',
    endDate: 'January 2019',
    description: 'A micro-services project management application\n'
    + 'which helps managers to keep track of resource allocation',
  },
  {
    isCurrentProject: false,
    name: 'PDV',
    role: 'Developer',
    startDate: 'February 2018',
    endDate: 'June 2018',
    description: 'Provider Search is a user interface that allows\n'
    + 'users to search for practitioners in the MasterFile. Built on the Single',
  },
  {
    isCurrentProject: false,
    name: 'DMTool',
    role: 'Developer',
    startDate: 'September 2017',
    endDate: 'February 2018',
    description: 'An internal tool which supports managers to keep track of resource allocation',
  },
];
