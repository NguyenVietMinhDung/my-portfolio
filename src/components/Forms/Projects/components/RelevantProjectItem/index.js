import RelevantProjectItem from './RelevantProjectItem';

export default RelevantProjectItem;
