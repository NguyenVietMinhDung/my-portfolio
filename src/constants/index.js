import * as actions from './actions';
import * as app from './app';
import * as navbar from './navbar';
import * as header from './header';
import * as footer from './footer';
import * as aboutMe from './aboutMe';
import * as experiences from './experiences';
import * as educations from './educations';
import * as projects from './projects';
import * as achievements from './achievements';

export default {
  actions,
  app,
  navbar,
  header,
  footer,
  aboutMe,
  experiences,
  educations,
  projects,
  achievements,
};
