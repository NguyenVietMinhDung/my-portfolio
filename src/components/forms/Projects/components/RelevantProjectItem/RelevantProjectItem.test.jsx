import React from 'react';
import { shallow } from 'enzyme';
import { items } from '../../../../../constants/projects';
import RelevantProjectItem from './RelevantProjectItem';

const defaultProps = items[0];

describe('<RelevantProjectItem />', () => {
  test('renders correctly', () => {
    const component = shallow(<RelevantProjectItem {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
});
