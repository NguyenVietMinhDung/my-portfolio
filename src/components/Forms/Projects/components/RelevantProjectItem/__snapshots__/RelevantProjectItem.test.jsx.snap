// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RelevantProjectItem /> renders correctly 1`] = `
<li
  className="relevant-project-item"
>
  <div
    className="name"
    data-aos="zoom-in"
    data-aos-duration="2000"
    data-aos-offset="100"
  >
    Textme
  </div>
  <div
    className="summary"
    data-aos="zoom-in"
    data-aos-duration="2000"
    data-aos-offset="100"
  >
    <div
      className="description"
    >
      A chat tool that helps customers can contact with bankers and versa vice
    </div>
    <div
      className="role"
    >
      Developer
    </div>
    <div
      className="duration"
    >
      <i
        className="lni-calendar"
      />
        
      September 2019 - May 2020
    </div>
  </div>
</li>
`;
